[KEYCLOAK-8234] Replace various fixed versioned strings with their parametrized counterparts

iankko opened the pull request:
Substitute fixed (current and previous) template  names and (current and previous) product names with their parametrized versions for simple future maintenance.

Also drop mention about JBoss EAP 7.0 image and templates, where appropriate, since these are deprecated already (see CLOUD-2699 / CLOUD-2678 for details)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[KEYCLOAK-8234] Parametrize mention of application templates
in the documentation, so various versions can be used

Also, use "[subs="verbatim,macros,attributes"]" as the default
for code sections in the documentation. Escape '<', '>', '+'
special characters, where necessary for proper rendering of
final documentation

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[KEYCLOAK-8234] Point the application templates installation step
to the new templates location

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[KEYCLOAK-8234] Replace various fixed versioned strings
with their parametrized version

Also drop mention about EAP 7.0 images / templates, since
these are deprecated (see CLOUD-2699 / CLOUD-2678 for details)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
@matthewhelmke @Crumby @pdrozd @stianst PTAL (once got a chance)

Thank you, Jan

stianst commented on topics/templates/document-attributes-product.adoc:
```diff
@@ -2,6 +2,7 @@
 :project_product: true
 :project_product_cd: true
 :project_name: Red Hat Single Sign-On
+:project_name_abbr: RH-SSO
```
Please remove project_name_abbr. We should always use "Red Hat Single Sign-On" in the documentation.

iankko replied:
@stianst `project_name_abbr` dropped, see https://github.com/keycloak/keycloak-documentation/pull/496/commits/63be10435495b24ade7703df904c75e59f89ab60 for details.

stianst commented on topics/templates/document-attributes-product.adoc:
```diff
@@ -11,6 +12,10 @@ ifeval::[{project_product_cd}==false]
 :project_name_full: Red Hat Single Sign-On
 :project_version: 7.3.0.GA
 :project_versionDoc: 7.3
+:project_previous_version: 7.2
```
I don't like the introduction of "project_previous_version" it also doesn't seem to be used anywhere.

iankko replied:
@stianst It was used in _topics/tutorials.adoc_:
```
[iankko@localhost openshift]$ grep -rHn 'project_previous_version' * | wc -l
6
```
but not that many times, it would deserve a dedicated asciidoc attribute. Will remove it.

iankko replied:
Dropped `project_previous_version` and refactored _Manual & Automatic DB Migration_ sections not to actually need it, see https://github.com/keycloak/keycloak-documentation/pull/496/commits/98cc71a8ba955182ff1629e1bc4719efe35597e8 for details.

stianst commented on topics/templates/document-attributes-product.adoc:
```diff
@@ -11,6 +12,10 @@ ifeval::[{project_product_cd}==false]
 :project_version: 7.3.0.GA
 :project_versionDoc: 7.3
+:project_previous_version: 7.2
+:project_previous_templates_version: sso-cd
```
Same as above. Doesn't seem to be used anywhere?

iankko replied:
The _project_previous_templates_version_ is again used in `topics/tutorials.adoc` 30-ty times this time:
```
[iankko@localhost openshift]$ grep -rHn 'project_previous_templates_version' * | wc -l
30
```

I would like to keep this one, if possible a bit (especially since the previous templates version is different across the products, and it will need to be changed again once 7.3 is GA).

iankko replied:
Dropped `project_previous_templates_version` and refactored _Manual & Automatic DB Migration_ sections not to actually need it, see https://github.com/keycloak/keycloak-documentation/pull/496/commits/98cc71a8ba955182ff1629e1bc4719efe35597e8 for details.

matthewhelmke reviewed (dismissed):
LGTM. Thank you for fixing the cause of the problem and not just the immediate issue. :+1:

stianst commented:
@iankko please address my comments and resolve conflicts

Commit:
[KEYCLOAK-8234] Drop the 'project_name_abbr' attribute

by substituting it with 'project_name' one, where previously referenced

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[KEYCLOAK-8234] Drop the 'project_previous_version' and 'project_previous_templates_version' Asciidoc attributes

Refactor the automatic & manual database migration sections not to be tied that much to a
specific, previous RH-SSO image version in order to these two attributes not to be needed

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
@matthewhelmke @stianst Fixed the conflicts, rebased against current master, dropped the requested Asciidoc attributes (intentionally kept the recently added commits unsquashed, so the changes WRT to the original PR are better visible).

PTAL (once got a chance)

Thank you, Jan

matthewhelmke reviewed (dismissed):
LGTM

stianst commented on openshift/topics/advanced_concepts.adoc:
```diff
@@ -49,17 +51,19 @@ See the https://access.redhat.com/documentation/en-us/jboss_enterprise_applicati
 .. Generate a CA certificate. Pick and remember the password. Provide identical password, when xref:signing-csr-with-ca-certificate[signing the certificate sign request with the CA certificate] below:
 +
-[subs="verbatim,macros"]
+[subs="verbatim,macros,attributes"]
```
Use this format everywhere you have code blocks (replacing bash with appropriate type):
`[source,bash,subs="attributes+"]`

See https://raw.githubusercontent.com/keycloak/keycloak-documentation/master/securing_apps/topics/oidc/keycloak-gatekeeper.adoc for examples

iankko replied:
@stianst Thanks for the suggestion, Stian! Used `[source,bash,subs="attributes+,macros+"]` form in order expressions like e.g. `<pass:quotes[_user-name_]>` would be expanded properly too.

Updated all occurrences to this new form.

stianst commented on openshift/topics/advanced_concepts.adoc:
```diff
@@ -1,40 +1,42 @@
 == Advanced Concepts
 
-These cover additional configuration topics, such as seting up keystores and a truststore for the RH-SSO server, creating an administrator account, an overview of available RH-SSO client registration methods, and guidance on configuring clustering.
+These cover additional configuration topics, such as seting up keystores and a truststore for the {project_name_full} server, creating an administrator account, an overview of available {project_name_full} client registration methods, and guidance on configuring clustering.
```
Use {project_name} throughout, not {project_name_full}

iankko replied:
Per suggestion replaced (mostly all of them) with `{project_name}`.

Commit:
[KEYCLOAK-8234] Address multiple issues:

* Whitelist .../templates/${resource} link for both RH-SSO 7.3 TP CD & RH-SSO 7.3 products,
* Replace '[subs="verbatim,macros,attributes"]' with '[source,bash,subs="attributes+,macros+"]'

  The 'macros+' field is needed in order also constructs like 'pass:quotes[_user-name_]' got
  expanded correctly,

* Replace deprecated 'oc env' command with its new "oc set env" equivalent,
* Replace deprecated 'oadm' command with its new "oc adm" equivalent,
* Escape selected https:// links, so they wouldn't end up rendered as <link xlink=...>

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[General] Recommend 7.2 or {project_version} as the proper RH-SSO
image versions to be used for new apps

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[General] Drop the duplicate occurrence of the 'Deployment Process' paragraph

Keep it in the 'Advanced Concepts' section, since that's the one, where it belongs
(see 7.2 guide for reference)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[General] Drop the duplicate occurrence of the '{project_name} Clients'
section and 'Limitations' section from the 'Get Started' book section

Keep them in the 'Advanced Concepts' section, since this is the one
where these two belongs (see 7.2 guide for reference)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[General] Drop the duplicate 'Creating Administrator Account for {project_name} Server'
paragraph from the 'Get Started' section

Keep the copy present in 'Advanced Concept's section, because it is more
updated & already present part of the book dedicated to advanced topics
(see 7.2 guide for reference)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

stianst committed:
Update introduction.adoc

iankko commented:
@stianst Thanks for updating `introduction.adoc` file & prompt review!

Dropped the _DO NOT MERGE_ label from this one. I still need to clean-up / sanitization of couple of sections in the _OpenShift_ documentation book here & there, but going to do that within another PR (since it's not related to the purpose of the original's [KEYCLOAK-8234](https://issues.jboss.org/browse/KEYCLOAK-8234) at all. Feel free to merge this one as convenient / appropriate for you.

Thanks!, Jan

matthewhelmke commented:
@iankko this LGTM. Can you resolve the conflicts? Once that is taken care of we can finally merge.

iankko commented:
@matthewhelmke Fixed the conflicts & rebased against current _master_.

matthewhelmke reviewed (dismissed):
Thank you!

matthewhelmke commented:
Any idea about why the build is failing, @iankko?

iankko commented:
Fixed the conflicts && rebased against current master.

@matthewhelmke Re: failing build, I have a suspicion the updated registry URL isn't a real final URL, causing issues, when testing. But let me dig further.

iankko commented:
@matthewhelmke Actually that invalid new registry page redirect was a red-herring. AFAICT Travis above failed because it failed to download some intermediary artifact from Central repo. Looks like increasing the timeout could fix that (will check a fix for this).

hmlnarik commented:
@iankko Could you please (yet again) resolve conflicts?

iankko commented:
@hmlnarik Conflicts resolved && rebased to current _master_ again (sorry for the delay).

matthewhelmke reviewed (approved):
Thank you
